c/main.c

```diff
@@ -0,0 +1,150 @@
+#include <deltachat.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+void handle_message(dc_context_t *context, int chat_id, int message_id) {
+  dc_chat_t *chat = dc_get_chat(context, chat_id);
+
+  if (dc_chat_get_type(chat) == DC_CHAT_TYPE_SINGLE) {
+    dc_msg_t *msg = dc_get_msg(context, message_id);
+
+    if (!msg) {
+      return;
+    }
+
+    dc_send_text_msg(context, chat_id, dc_msg_get_text(msg));
```

Comment:
to avoid a memory leak, the result of `dc_msg_get_text()` needs to be freed:

```suggestion
    char* text = dc_msg_get_text(msg);
    dc_send_text_msg(context, chat_id, text);
    dc_str_unref(text);
```